### PR TITLE
use controller runtime impersonation client for member provider

### PR DESCRIPTION
**Pull request by zreigz**
**What this PR does / why we need it**: use controller runtime impersonation client for member provider

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

```release-note
NONE
```

### CODE REVIEW

File: api/pkg/handler/v1/user/user_test.go [hunk 1]
```diff
@@ -393,7 +388,7 @@ func TestDeleteUserFromProject(t *testing.T) {
 			res := httptest.NewRecorder()
 			kubermaticObj := []runtime.Object{}
 			kubermaticObj = append(kubermaticObj, tc.ExistingKubermaticObjs...)
-			ep, clients, err := test.CreateTestEndpointAndGetClients(tc.ExistingAPIUser, nil, []runtime.Object{}, []runtime.Object{}, kubermaticObj, nil, nil, hack.NewTestRouting)
+			ep, _, err := test.CreateTestEndpointAndGetClients(tc.ExistingAPIUser, nil, []runtime.Object{}, []runtime.Object{}, kubermaticObj, nil, nil, hack.NewTestRouting)
```
Comment (lsviben) [inline api/pkg/handler/v1/user/user_test.go]: I think you can use `test.CreateTestEndpoint` here

File: api/pkg/handler/v1/user/user_test.go [hunk 2]
```diff
@@ -384,7 +380,6 @@ func TestDeleteUserFromProject(t *testing.T) {
 			ExistingAPIUser:              *genAPIUser("admin", "admin@acme.com"),
 			ExpectedResponse:             `{}`,
 			ExpectedBindingIDAfterDelete: test.GenBinding("plan9-ID", "bob@acme.com", "viewers").Name,
```
Comment (lsviben) [inline api/pkg/handler/v1/user/user_test.go]: you can also remove these `ExpectedBindingIDAfterDelete`

File: api/pkg/handler/v1/user/user_test.go [hunk 3]
```diff
@@ -612,7 +581,7 @@ func TestEditUserInProject(t *testing.T) {
 			res := httptest.NewRecorder()
 			kubermaticObj := []runtime.Object{}
 			kubermaticObj = append(kubermaticObj, tc.ExistingKubermaticObjs...)
-			ep, clients, err := test.CreateTestEndpointAndGetClients(tc.ExistingAPIUser, nil, []runtime.Object{}, []runtime.Object{}, kubermaticObj, nil, nil, hack.NewTestRouting)
+			ep, _, err := test.CreateTestEndpointAndGetClients(tc.ExistingAPIUser, nil, []runtime.Object{}, []runtime.Object{}, kubermaticObj, nil, nil, hack.NewTestRouting)
```
Comment (lsviben) [inline api/pkg/handler/v1/user/user_test.go]: you can also use `test.CreateTestEndPoint` here and remove the `ExpectedBindingAfterUpdate`

File: api/pkg/handler/v1/user/user_test.go [hunk 4]
```diff
@@ -920,7 +862,7 @@ func TestAddUserToProject(t *testing.T) {
...
```
Comment (lsviben) [inline api/pkg/handler/v1/user/user_test.go]: here as above, and you can remove `ExpectedBindingAfterInvitation`

**Review by lsviben — CHANGES_REQUESTED** [review]
Looks good, just some cleanup remarks for tests

**Comment by zreigz** [comment]
@lsviben thanks for the review :+1: 
PTAL

**Pull request merged**